src/CoreInstaller.php

```diff
@@ -76,12 +76,34 @@ public function install($source, $destination)
                 continue;
             }
 
-            copy($item, $destinationFile);
+            $this->installFile($item, $destinationFile);
             $this->gitIgnore->addEntry('/' . $iterator->getSubPathName());
         }
     }
 
     /**
+     * @param string $item
+     * @param string $destinationFile
+     */
+    protected function installFile($item, $destinationFile)
+    {
+        // Hardlinks require to be on same Filesystem/Blockdevice
+        list($source_dev) = stat($item);
```

Comment:
I think it would more performant and more readable if the check for same filesystem was performed only once at the start of the install process. If not on the same FS we can print a debug message stating that we are falling back to symlink.